### DISCUSSION TITLE
.NET 6 and Single Project Solutions

### DISCUSSION THREAD

**Discussion by robloo** [unoplatform/uno#6096] [Q&A] [answered]
.NET 6 with MAUI is finally unifying the various platform heads into one single project. See the example for MAUI here:

![](https://devblogs.microsoft.com/dotnet/wp-content/uploads/sites/10/2020/05/maui-02.png)

I wonder what is the Uno Platform plan for adopting a similar approach? I've seen this discussed some years ago and it would be great to drop all the various project heads. All the heads create additional complexity and require a lot of time to, for example, update dependencies (without a bot). My assumption is the Uno Platform have been working with Microsoft on Uno's requirements to support this?

Other links & discussions:
 * https://devblogs.microsoft.com/dotnet/introducing-net-multi-platform-app-ui/#single-project-developer-experience
 * https://github.com/dotnet/maui/discussions/70

**Comment by francoistanguay** [comment, marked as answer]
Yes, we've been actively working on this and would've shipped something already if it wasn't for all the VS issues. We're hoping the next VS preview will solve many of them and we'll eventually be able to turn the switch on, knowing that we do have additional heads that aren't supported by "Single Project" yet. So it's a "partial" yes.

**Reply by robloo** [reply]
Thanks for confirming there is movement in this area. I look forward to trying it out at least for the platforms where it's supported by Xamarin (iOS, Droid, MacOS). If Wasm and Linux need to have separate heads until Xamarin gets on board with those platforms it is still a huge improvement. 

I understand the VS issues, there do seem to be quite a few at the moment. Still, look forward to seeing what comes out of this later this year.

**Comment by HavenDV** [comment]
At the moment, you can try the single project solution within a separate repo - https://github.com/HavenDV/Uno.Sdk. This is a workload similar to tizen and this delivers`net7.0-webassembly;net7.0-gtk;net7.0-linux` to distinguish WebAssembly/Skia.Gtk/Skia.Linux.Framebuffer respectively. Skia.Wpf is implemented within `net7.0-windows` (this is an implicit indication for `net7.0-windows7`, which is different from the WinUI TargetFramework - `net7.0-windows10.x.x.x`).
Check the documentation for more details.
I would be glad to receive any feedback
